### PR TITLE
drop index through compact

### PR DESCRIPTION
**Pull request by bright-starry-sky**
When `DROP INDEX` command is complete, At this time, only the metadata of index was removed.
Then, index data needs to be deleted through the compact in storage layer.

### CODE REVIEW

File: src/storage/test/CompactionTest.cpp [hunk 1]
```diff
@@ -155,11 +155,54 @@ void mockTTLDataNotExpired(kvstore::KVStore* kv) {
...
 
+void mockIndexData(kvstore::KVStore* kv) {
+    auto tagId = 3001;
+    auto edgeType = 101;
+    IndexValues values;
+    values.emplace_back(nebula::cpp2::SupportedType::STRING, "col1");
+    for (PartitionID partId = 0; partId < 3; partId++) {
+        std::vector<kvstore::KV> data;
+        for (auto vertexId = partId * 10; vertexId < (partId + 1) * 10; vertexId++) {
+            // one tag data, the record data will never expire
```
Comment (panda-sheep) [inline src/storage/test/CompactionTest.cpp]: invalid comments ?

Comment (bright-starry-sky) [reply src/storage/test/CompactionTest.cpp]: > invalid comments ?

Fixed , Thanks reminding.

**Comment by panda-sheep** [comment]
Excellent！

**Comment by dangleptr** [comment]
Please check the failed UT . @bright-starry-sky

**Comment by bright-starry-sky** [comment]
> Please check the failed UT . @bright-starry-sky

Yes, I'm checking now , asyncMultiPut error leads the test case failing.

**Comment by bright-starry-sky** [comment]
> > Please check the failed UT . @bright-starry-sky
> 
> Yes, I'm checking now , asyncMultiPut error leads the test case failing.

It is unrelated with this PR. This is an unstable testcase. I will improve this failing case at later.

**Pull request merged**